src/commit.c

```diff
@@ -104,7 +104,7 @@ int git_commit_create_from_callback(
 
 	if (update_ref != NULL) {
 		error = git_reference__update_for_commit(
-			repo, ref, update_ref, id, committer, "commit");
```

Comment:
It may be that this is the way Git does it, but this seems weird.  I would expect the reflog to move in accordance with the `committer` that was passed in, not the committer that is specified as the repository's default...